### DISCUSSION TITLE
Reducing outliers without changing topic representations.

### DISCUSSION THREAD

**Discussion by rcote** [MaartenGr/BERTopic#1198] [Q&A]
Hi. I have reduced the outliers in my model using model.reduce_outliers(docs, topics). Is it possible to change the document topic assignments in the model itself but without changing the topic representations? Is it what model._update_topic_size(documents) does (it is mentioned here https://github.com/MaartenGr/BERTopic/issues/485 but does not seem to be documented anymore)?

Note that I want to do that so all further analysis and visualizations will use the new assignments, but without changing the topic representations. I do not want to change the topic representation because I fear that the addition of new documents to the topics make the topic representations less crisp and clear due to the additional, possibly less related, documents.

Note also that the documentation here "https://maartengr.github.io/BERTopic/getting_started/outlier_reduction/outlier_reduction.html#update-topics" says that it is possible to do what I want but without explaining how to do it...

Thanks for your help!

**Comment by MaartenGr** [comment]
> Is it what model._update_topic_size(documents) does (it is mentioned here https://github.com/MaartenGr/BERTopic/issues/485 but does not seem to be documented anymore)?

I believe it is indeed the way to update the assignments here, although it is been a while since I used that as it is a private function on purpose. You could try it out and see if that works but I cannot guarantee it does not break things elsewhere.

**Reply by rcote** [reply]
I tried model._update_topic_size(document) and it seems to work as I wanted. Thanks!
